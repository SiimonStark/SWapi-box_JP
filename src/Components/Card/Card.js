import React from 'react';
import PropTypes from 'prop-types';

const Card = ({person}) => {
  let {name, species, homeworld, population, favorite} = person;
  return (
    <section className="Card">
      <h4>{name}</h4>
      <p>Species: {species}</p>
      <p>Homeworld: {homeworld}</p>
      <p className="home-pop">Homeworld Population:</p>
      <p>{population}</p>
      <button className={`fave-${favorite}`}>
        <i class="fas fa-star" />
      </button>
    </section>
  );
}

Card.propTypes ={
  person: PropTypes.object.isRequired
}

export default Card;