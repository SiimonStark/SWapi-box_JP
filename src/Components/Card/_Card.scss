.Card {

  .fade {
    position: absolute;
    width: 100%;
    min-height: 60vh;
    left: 0;
    top: 0;
    background-image: linear-gradient(0deg, transparent, black 90%);
    z-index: 1;
  }

  flex-wrap: wrap;
  width: 100%;
  height: 100%;
  margin: auto;
  padding: 5%;
}

.card__content {
  padding: 20px;
  position: absolute;
  z-index: 3;
  border: $OGYellow 5px ridge;
  border-radius: 20px;
  background: linear-gradient(340deg, slategray 55%, silver 85%);

  h4 {
    font-size: 32px;
    margin: 0 5px 20px 5px;
    text-decoration-line: underline;
  }
  p {
    margin: 5px;
    font-size: 22px;
  }
}