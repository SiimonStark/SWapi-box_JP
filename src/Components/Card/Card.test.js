import React from 'react';
import {shallow} from 'enzyme';
import Card from './Card';

let mockPerson = 'chewwy'

describe('Card', ()=>{
  let wrapper;

  it('should match snapshot', ()=>{
    wrapper = shallow(<Card person={mockPerson}/>);
    expect(wrapper).toMatchSnapshot();
  });
});