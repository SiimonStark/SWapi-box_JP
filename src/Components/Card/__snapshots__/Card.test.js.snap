// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
<section
  className="Card"
>
  <h4 />
  <p>
    Species: 
  </p>
  <p>
    Homeworld: 
  </p>
  <p
    className="home-pop"
  >
    Homeworld Population:
  </p>
  <p />
  <button
    className="fave-undefined"
  >
    <i
      class="fas fa-star"
    />
  </button>
</section>
`;
